src/services/SlotService.ts

```diff
@@ -96,5 +112,34 @@ export default class SlotService implements IService<Slot> {
       ));
     }
   }
+
+
+
+  public async disableSlotsByRoomId(
+    roomId: number,
+    roomData: RoomUpdateBodyValidator,
+  ): Promise<void> {
+
+    const repo = this.getRepository();
+
+    const slots = await this.getAllSlotsByRoomId(roomId);
+
+    const slotsIds = [];
+
+    for (let i = 0; i < slots.length; i += 1) {
```

Comment:
Using `for` loops as this should be the last resort, it is too imperative and harder to read, you can achieve the same result with a more functional/declarative approach:

```js
const slotIds = slots.map(s => s.id)
```

Furthermore, this is so short that you can just use it in line 138 directly:

```js
        ? bookingService.deleteBookingsBySeveralSlotIds(slots.map(s => s.id))
```